### DISCUSSION TITLE
Question relating to pfsense_alias and pfsense_rule modules.

### DISCUSSION THREAD

**Discussion by hendrik43** [pfsensible/core#143] [General]
Good afternoon @opoplawski 

Thank you for developing this module. I have a couple of questions, and I hope this is the appropriate place to ask. If not, kindly point me in the right direction.

First, regarding the pfsense_alias module, is there a way to modify an existing alias without overwriting its current entries? When I attempted this, it replaced all the existing data with the new information.

Second, with the pfsense_rule module, when using the after: option, is it possible to specify a separator instead of a name or description?

Lastly, I’ve encountered a similar issue with firewall rules where attempting to edit an existing rule results in it being completely overwritten.

Thank you for your assistance.

**Comment by opoplawski** [comment]
I'm not entirely sure what kind of behavior you are expecting with the alias and rule modules.  As written, they are intended to fully define the alias or rule as described in the module parameters.  Are you expecting to be able to only modify specific fields or add entries?  Could you write an example play and give before and after definitions of the aliases/rules?

No, you can't reference separators in a rule's before: or after: field.   The separator locations are based on the rules, not the other way around.

**Comment by hendrik43** [comment]
To add and remove an alias works amazing, my question referes to the when clause "edit"

When I use the playbook to manage an alias that blocks IP addresses, and there are 10 IP addresses currently listed, selecting when: add_or_edit_or_remove == "edit" causes the playbook to overwrite those 10 addresses with a single new entry. Is there a way to prevent this from happening?

Here’s the relevant section of my playbook:

---
- name: Manage pfSense aliases
  hosts: Example
  gather_facts: false
  become: yes

  vars:
    ansible_python_interpreter: /usr/local/bin/python3.11

  tasks:
    - name: Add alias on pfSense
      pfsensible.core.pfsense_alias:
        name: "{{ alias_name }}"
        address: "{{ alias_ip_address }}"
        descr: "{{ alias_description }}"
        type: "{{ alias_type }}"
        state: present
      when: add_or_edit_or_remove == "add"

    - name: Update alias on pfSense
      pfsensible.core.pfsense_alias:
        name: "{{ alias_name }}"
        address: "{{ alias_ip_address }}"      # New IP address
        type: "{{ alias_type }}"                        # Use same type as before
        state: present                                      # Alias must remain present
      when: add_or_edit_or_remove == "edit"

    - name: Remove alias on pfSense
      pfsensible.core.pfsense_alias:
        name: "{{ alias_remove_name }}"
        state: absent
      when: add_or_edit_or_remove == "remove"

**Comment by opoplawski** [comment]
So, I can see the appeal, but I think it would be a pain to implement it.  You would also either need to have additional states (add, remove) to add / remove items, or another parameter with similar options to control it.  I would entertain a pull request to do that, but since I don't have a use case for it at the moment I'm afraid I just don't have the time to work on it.
